plugins/toolkit/jetbrains-ultimate/tst/software/aws/toolkits/jetbrains/services/rds/auth/IamAuthTest.kt

```diff
@@ -160,6 +174,32 @@ class IamAuthTest {
             .doesNotStartWith("https://")
     }
 
+    @Test
+    fun `Handle Sso authentication no token present`() {
+        val noTokenCredential = RuleUtils.randomName()
+        credentialManager.addCredentials(ProfileCredentialsIdentifierSso(noTokenCredential, noTokenCredential, "", CredentialType.SsoProfile))
+        credentialManager.addSsoProvider(ProfileSsoSessionIdentifier(noTokenCredential, "", "", setOf("")))
+        val conneciton = buildConnection(hasCredentials = true, credentialId = "profile:" + noTokenCredential)
+        mockkObject(ToolkitAuthManager.Companion)
+        val defaultToolkitAuthManager = mockk<DefaultToolkitAuthManager>()
+        val profileSsoManagedBearerSsoConnection = mockk<ProfileSsoManagedBearerSsoConnection>()
+        every { ToolkitAuthManager.getInstance() } returns defaultToolkitAuthManager
+        every { defaultToolkitAuthManager.getOrCreateSsoConnection(any(UserConfigSsoSessionProfile::class)) } returns profileSsoManagedBearerSsoConnection
+        val bearerTokenProvider = mockk<BearerTokenProvider>()
+        // We don't have the token
+        every { bearerTokenProvider.state() } returns BearerTokenAuthState.NOT_AUTHENTICATED
+        every { bearerTokenProvider.reauthenticate() } returns Unit
+        val toolkitBearerTokenProvider = ToolkitBearerTokenProvider(bearerTokenProvider)
+        every {
+            profileSsoManagedBearerSsoConnection.getConnectionSettings()
+        } returns TokenConnectionSettings(toolkitBearerTokenProvider, AwsRegion("us-east-1", "", ""))
+
+        val connection = iamAuth.handleSsoAuthentication(projectRule.project, conneciton)
+        assertThat(connection).isNotNull
+        // We need to request the browser login once.
+        verify(exactly = 1) { bearerTokenProvider.reauthenticate() }
```

Comment:
? shouldn't this only happen if the user clicks on the login action?

Reply:
We can't test for click. So, we are testing the method that is called after clicking. We are calling `handleSsoAuthentication` which will be called after the user clicks on the login action. `handleSsoAuthentication` in turns calls for reauthentication.

Reply:
test case fixed